snuba/subscriptions/scheduler.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from typing import Generic, Iterator, Tuple, TypeVar
+
+from snuba.utils.types import Interval
+from snuba.subscriptions.types import TTimestamp
+
+
+TTask = TypeVar("TTask")
+
+
+class Scheduler(ABC, Generic[TTimestamp, TTask]):
```

Comment:
In practice, this will be `Scheduler[datetime, Tuple[UUID, Subscription]]` or similar.

Reply:
thanks this is useful.